app/services/planning_application_filter.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class PlanningApplicationFilter
+  include ActiveModel::Model
+
+  attr_accessor :filter_options, :planning_applications, :user
+
+  def results
+    records_matching_reference || []
+  end
+
+  def filter_types
+    filter_options.reject(&:empty?)
+  end
+
+  def filtered_filter_types
+    filter_types.map { |x| x == "to_be_reviewed" ? "awaiting_correction" : x }
+  end
+
+  private
+
+  def records_matching_query
+    records_matching_reference.presence
+  end
+
+  def records_matching_reference
+    filtered_filter_types.map { |filter| planning_applications.send(filter) }.flatten
```

Comment:
Is this doing more queries than necessary?

Could we not load all the planning applications in memory and then filter them from this point. I think this will send a separate query for each different status

Reply:
Yeah think this was a left over from me trying to do it another stupider way